examples/pipeline_tacotron2/text/text_preprocessing.py

```diff
@@ -63,23 +64,102 @@
+        # '[HH][AH][L][OW] [W][ER][L][D]!'
+        sent = _phonemizer(sent, lang='en_us')
+
+        # ['[HH]', '[AH]', '[L]', '[OW]', ' ', '[W]', '[ER]', '[L]', '[D]', '!']
+        ret = re.findall(_phone_symbols_re, sent)
+
+        # ['HH', 'AH', 'L', 'OW', ' ', 'W', 'ER', 'L', 'D', '!']
+        ret = [r.replace("[", "").replace("]", "") for r in ret]
+
+        return ret
+    else:
+        raise ValueError(f"The `phonemizer` {phonemizer} is not supported."
+                         "Supported `symbol_list` includes `'DeepPhonemizer'`.")
+
+
+def text_to_sequence(sent: str,
+                     symbol_list: Union[str, List[str]] = "english_characters",
+                     phonemizer: Optional[str] = "DeepPhonemizer",
+                     checkpoint: Optional[str] = "./en_us_cmudict_forward.pt",
+                     cmudict_root: Optional[str] = "./") -> List[int]:
     r'''Converts a string of text to a sequence of IDs corresponding to the symbols in the text.
 
-      Args:
-        sent (str): The input sentence to convert to a sequence.
+        Args:
+            sent (str): The input sentence to convert to a sequence.
+            symbol_list (str or List of string, optional): When the input is a string, available options include
+                "english_characters" and "english_phonemes". When the input is a list of string, ``symbol_list`` will
+                directly be used as the symbol to encode. (Default: "english_characters")
+            phonemizer (str, optional): The phonemizer to use. Only used when ``symbol_list`` is "english_phonemes".
+                Available options include "DeepPhonemizer". (Default: "DeepPhonemizer")
+            checkpoint (str, optional): The path to the checkpoint of the phonemizer. Only used when ``symbol_list`` is
+                "english_phonemes". (Default: "./en_us_cmudict_forward.pt")
+            cmudict_root (str, optional): The path to the directory where the CMUDict dataset is found or downloaded.
+                Only used when ``symbol_list`` is "english_phonemes". (Default: "./")
+
+        Returns:
+            List of integers corresponding to the symbols in the sentence.
 
-      Returns:
-        List of integers corresponding to the symbols in the sentence.
+        Examples:
+            >>> text_to_sequence("hello world!", "english_characters")
+            [19, 16, 23, 23, 26, 11, 34, 26, 29, 23, 15, 2]
+            >>> text_to_sequence("hello world!", "english_phonemes")
+            [54, 20, 65, 69, 11, 92, 44, 65, 38, 2]
     '''
+
     sent = unidecode(sent)  # convert to ascii
     sent = sent.lower()  # lower case
     sent = normalize_numbers(sent)  # expand numbers
     for regex, replacement in _abbreviations:  # expand abbreviations
         sent = re.sub(regex, replacement, sent)
     sent = re.sub(_whitespace_re, ' ', sent)  # collapse whitespace
 
+    if isinstance(symbol_list, list):
+        symbols = symbol_list.copy()
```

Comment:
At a glance, this `copy` seems unnecessary. Please verify it.

Reply:
That's right, I've removed the copy [here](https://github.com/pytorch/audio/pull/1668/files#diff-2b633e14bbda0eeaa9126220bbcc9782404e90d7cf7a0a4b2ab1be29dda27e56R158)